cranelift-codegen/meta/src/isa/x86/encodings.rs

```diff
@@ -838,6 +841,18 @@ pub fn define(
     e.enc_i32_i64(fill, rec_fillSib32.opcodes(vec![0x8b]));
     e.enc_i32_i64(regfill, rec_regfill32.opcodes(vec![0x8b]));
 
+    // No-op fills, created by late-stage redundant-fill removal.
+    for &ty in &[I64, I32, I16, I8] {
+        e.enc64_rec(fill_nop.bind(ty), rec_fillnull, 0);
+        e.enc32_rec(fill_nop.bind(ty), rec_fillnull, 0);
+    }
+    e.enc64_rec(fill_nop.bind(B1), rec_fillnull, 0);
+    e.enc32_rec(fill_nop.bind(B1), rec_fillnull, 0);
+    for &ty in &[F64, F32] {
```

Comment:
I just saw that there might be a chance we can use `ValueType::all_lane_types` here to iterate on all the types; it will add bindings for other boolean types, in addition to what you already have, which could be a good thing.

Reply:
On investigation, that seems not to be possible, because we need different recipes, `rec_fillnull` vs `rec_ffillnull`, for the int/bool vs float cases.  And those two recipes can't be combined because they reference different register sets (`GPR` vs `FPR`).